framework/src/variables/MooseVariableInterface.C

```diff
@@ -31,11 +33,22 @@ MooseVariableInterface<T>::MooseVariableInterface(const MooseObject * moose_obje
   // Try the scalar version first
   std::string variable_name = parameters.getMooseType(var_param_name);
   if (variable_name == "")
+  {
+    auto vec = parameters.getVecMooseType(var_param_name);
+
+    // Catch the (unlikely) case where a user specifies variable = ''
+    // in their input file. This could happen if e.g. something goes
+    // wrong with dollar bracket expression expansion.
+    if (vec.empty())
+      mooseError("Error retrieving variable name! Did you set variable = '' by accident?");
```

Comment:
We're trying to make errors a little less opaque (lol - that's actually the function of _this_ PR!): would you mind adding as much context as you can to this error statement?

Reply:
Hmm... maybe we could print `name` of the passed in MooseObject's `InputParameters`?

Reply:
Just do what you can - any amount of context will help